File: planning-idsdp-rms-rpm-backend/src/main/java/com/ibcs/idsdp/rpm/services/ResearcherProposalActionPlanService.java
package com.ibcs.idsdp.rpm.services;

import com.ibcs.idsdp.rpm.model.domain.ResearcherProposalActionPlan;
import com.ibcs.idsdp.rpm.web.dto.request.ResearcherProposalActionPlanRequestDto;
import com.ibcs.idsdp.rpm.web.dto.response.ResearcherProposalActionPlanResponseDto;
import com.ibcs.idsdp.util.Response;

import java.util.List;

public interface ResearcherProposalActionPlanService {

    public Response<ResearcherProposalActionPlanResponseDto> getByResearcherProposalId(Long id);

    public Response<ResearcherProposalActionPlanResponseDto> saveList(List<ResearcherProposalActionPlanRequestDto> request);

    public Response<ResearcherProposalActionPlanResponseDto>  saveList2(List<ResearcherProposalActionPlanRequestDto> actionPlans);

}

File: planning-idsdp-rms-frontend/src/app/main/modules/rpm/services/researcher-proposal-action-plan.service.ts
import {HttpClient} from '@angular/common/http';
import {Injectable} from '@angular/core';
import {CrudRequestService} from 'app/main/core/services/crud-request.service';
import {Observable} from "rxjs";
import {CoreResponse} from "../../../core/models/response";
import {
    GET_BY_RESEARCHER_PROPOSAL_ID,
    RESEARCHER_PROPOSAL_ACTION_PLAN,
    SAVE_LIST,
    SAVE_LIST2
} from "../contants/researcher-proposal-action-plan.constant";
import {ResearcherProposalActionPlan} from "../models/ResearcherProposalActionPlan";

@Injectable({
  providedIn: 'root'
})
export class ResearcherProposalActionPlanService extends CrudRequestService {


  constructor(private http: HttpClient) {
      super(http, RESEARCHER_PROPOSAL_ACTION_PLAN);
  }

    /**
     * Get by Researcher Proposal Id
     * @param id
     */
    getByResearcherProposalId(id: number): Observable<CoreResponse<ResearcherProposalActionPlan>> {
        return this.http.get<CoreResponse<ResearcherProposalActionPlan>>(this._BASE_URL + GET_BY_RESEARCHER_PROPOSAL_ID + "/" + id);
    }

    saveList(data: ResearcherProposalActionPlan[]): Observable<CoreResponse<ResearcherProposalActionPlan>> {
        return this.http.post<CoreResponse<ResearcherProposalActionPlan>>(this._BASE_URL + SAVE_LIST2, data);
    }


}
